baseline/baseline.py

```diff
@@ -0,0 +1,195 @@
+
+
+def input_sample_rate() -> int:
+    """
+    Returns:
+        One of the following values: [16000, 22050, 44100, 48000].
+            To avoid resampling on-the-fly, we will query your model
+        to find out what sample rate audio to provide it.
+    """
+    return RandomProjectionMelEmbedding.sample_rate
+
+
+def load_model(model_file_path: str, device: str = "cpu") -> Any:
+    """
+    In this baseline, we don't load anything from disk.
+    Args:
+        model_file_path: Load model checkpoint from this file path.
+            device: For inference on machines with multiple GPUs,
+            this instructs the participant which device to use. If
+            “cpu”, the CPU should be used (Multi-GPU support is not
+            required).
+    Returns:
+        Model
+    """
+    return RandomProjectionMelEmbedding().to(device)
+
+
+def get_audio_embedding(
+    audio: Tensor,
+    model: Any,
+    hop_size_samples: int,
+    batch_size: Optional[int] = None,
+    center: bool = True,
+) -> Tuple[Dict[int, Tensor], Tensor]:
+    """
+    Args:
+            audio: n_sounds x n_samples of mono audio in the range
+            [-1, 1]. We are making the simplifying assumption that
+            for every task, all sounds will be padded/trimmed to
+            the same length. This doesn’t preclude people from
+            using the API for corpora of variable-length sounds;
+            merely we don’t implement that as a core feature. It
+            could be a wrapper function added later.
+            model: Loaded model, in PyTorch or Tensorflow 2.x. This
+        should be moved to the device the audio tensor is on.
+        hop_size: Number of audio samples between adjacent frames
+            batch_size: The participants are responsible for estimating
+        the batch_size that will achieve high-throughput while
+        maintaining appropriate memory constraints. However,
+        batch_size is a useful feature for end-users to be able to
+        toggle.
+            center: If True, the timestamps correspond to the center
+            of each analysis window. center=True will be used for
+            all evaluation tasks.
+    Returns:
+            ({embedding_size: Tensor}, list(frame timestamps)) where
+            embedding_size can be any of [4096, 2048, 512, 128,
+            20].  Tensor is float32 (or signed int for 20-dim),
+            n_sounds x n_frames x dim.
+
+    """
+    assert audio.ndim == 2
+    model = model.to(audio.device)
+
+    # Implement batching of the audio
+    if batch_size is None:
+        # Here we just pick a sensible default batch size
+        batch_size = 512
+    dataset = torch.utils.data.TensorDataset(audio)
+    loader = torch.utils.data.DataLoader(
+        dataset, batch_size=batch_size, shuffle=False, drop_last=False
+    )
+
+    if center:
+        timestamps = (
+            torch.range(0, audio.shape[1], hop_size_samples) / input_sample_rate()
```

Comment:
This is minor, so we can ignore if you want -- but I'm getting deprecation warnings for range. Suggests to update to arange. Works essentially the same, some minor differences. The epsilon on end is suggested by pytorch to handle float errors, this crops up when hop size goes into the length of audio an even number of times.
```
timestamps = torch.arange(0, audio.shape[1] + 1e-8, hop_size_samples, dtype=torch.float)
timestamps = timestamps / input_sample_rate()
```

Reply:
+1